docs/eventsource-deployment-strategies.md

```diff
@@ -0,0 +1,49 @@
+# EventSource Deployment Strategies
+
+EventSource controller creates a k8s deployment for each EventSource object to
+watch the events. Some of the event source types do not allow multiple live
+clients with same attributes (i.e. multiple clients with same `clientID`
+connecting to a NATS server), or multiple event source PODs will generate
+duplicated events to downstream, so the deployment strategy and replica numbers
+are different for different event sources.
+
+## Rolling Update Strategy
+
+`Rolling Update` strategy is applied to the following EventSource types:
```

Comment:
Please carefully review these types @VaibhavPage

Reply:
AMQP and SQS should go under `Recreate`

Reply:
Done for AMQP.

SQS can stay in `Rolling Update`. Multiple SQS clients listening to the same queue for load balancing is ok.

Reply:
Reason for that:

1. We delete the the message after sending to EventBus; 
2. We set `VisibilityTimeout:   aws.Int64(120), // 120 seconds`, after reading a message, SQS makes sure it will be invisible for 120 seconds.